common/auto-updater/http-downloader.cpp

```diff
@@ -0,0 +1,186 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+
+
+#ifdef ENABLE_RS_AUTO_UPDATER
```

Comment:
Not sure I like this -- Maybe it should be in the header, and the .cpp shouldn't even be added by CMake?

What happens when someone turns auto-updater off and then some files include the header and use it? Have we tried?

Reply:
This design was made by request.
There are dummy function compiling when the flag is off.
It's been tested.